dcos/http.py

```diff
@@ -180,3 +220,29 @@ def delete(url, **kwargs):
 def silence_requests_warnings():
     """Silence warnings from requests.packages.urllib3.  See DCOS-1007."""
     requests.packages.urllib3.disable_warnings()
+
+
+def _get_http_auth_credentials(response):
+    """Get username/password for HTTP Basic Auth
+
+    :param response: requests.response
+    :type response: requests.Response
+    :returns: process return code
+    :rtype: int
+    """
+
+    parsed_url = urlparse(response.url)
+    hostname = parsed_url.hostname
+    if response.headers.get('www-authenticate') == 'Basic realm="Restricted"':
```

Comment:
`It is also special in that the server could have sent the same header multiple times with different values, but requests combines them so they can be represented in the dictionary within a single mapping, as per RFC 7230`